frontend/src/context/AuthProvider/AuthProvider.tsx

```diff
@@ -24,15 +25,42 @@ export const AuthContext = createContext<AuthContextT>({} as AuthContextT);
 
 export default function AuthProvider({ children }: { children: ReactNode }) {
   const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(true);
+
+    useEffect(() => {
+        const initializeAuth = async () => {
```

Comment:
I don't understand why you are creating a function and then calling it within the useEffect hook? Wouldn't be simpler to have the initializeAuth code directly inside the hook?

Reply:
else you would need to do something like the code snippet below ( because of its async nature) 
if done without a function

(async () => {
...
})();


having a function and calling it makes it a bit more readable imo